packages/explorerkit-server/src/server.ts

```diff
@@ -343,8 +358,35 @@ async function decodeInstruction(instruction: Instruction): Promise<Instruction>
   };
 }
 
-function postProcessDecodedInstruction(decodedInstructionData: any, decodedInstructionDataWithTypes: any): any {
-  if (decodedInstructionData === null || decodedInstructionDataWithTypes === null) {
+function decodeProgramError(programError: ProgramError): ProgramError {
+  if (!programError.errorCode) {
+    return programError
+  }
+
+  const programId = programError.programId;
+  const parser = parsersCache.get<SolanaFMParser>(programId);
+
+  if (!parser) {
+    return programError;
+  }
+
+  const hexErrorCode = `0x${programError.errorCode.toString(16)}`;
+  const errorParser = parser.createParser(ParserType.ERROR) as ErrorParserInterface;
+  const parsedError = errorParser.parseError(hexErrorCode);
+
+  if (!parsedError) {
+    return programError;
+  }
+
+  return {
+    ...programError,
+    decodedData: parsedError.data,
+    name: parsedError.name,
+  }
+}
+
+function postProcessDecodedInstruction(decodedInstructionData?: any, decodedInstructionDataWithTypes?: any): any {
+  if (!decodedInstructionData || !decodedInstructionDataWithTypes) {
```

Comment:
NIT: I know this is not something we currently have much of here, but any chance you could add some basic tests to make sure this full flow works? This is becoming more an more of a critical production dependency

Reply:
I didn't a slight refactor of explorer-kit server, and also added a couple of unit tests for specific critical pieces. It should be a little easier to maintain now.